I'm down to try the tests myself, but I can't get single file tests running.

`python3 -m unittest discover mpf/tests` worked to get me 859 passing (2 skips)
but trying the single file test with `mpf test mpf/tests/platforms/fast.py` first gave me a kivy missing error (havent needed it yet since not running mc) and after installing that, now gives an issue importing: `mpf.file_interfaces.yaml_roundtrip`

```
 Traceback (most recent call last):
   File "/home/knight/mpf80env/bin/mpf", line 8, in <module>
     sys.exit(run_from_command_line())
   File "/home/knight/workspace/mpf/mpf/commands/__init__.py", line 194, in run_from_command_line
     CommandLineUtility(path).execute()
   File "/home/knight/workspace/mpf/mpf/commands/__init__.py", line 174, in execute
     _module = import_module('mpf.commands.%s' % command)
   File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
     return _bootstrap._gcd_import(name[level:], package, level)
   File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
   File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
   File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
   File "<frozen importlib._bootstrap_external>", line 848, in exec_module
   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
   File "/home/knight/workspace/mpf/mpf/commands/test.py", line 12, in <module>
     from mpf.tests.MpfDocTestCase import MpfDocTestCase, MpfDocTestCaseNoFakeGame
   File "/home/knight/workspace/mpf/mpf/tests/MpfDocTestCase.py", line 11, in <module>
     from mpf.file_interfaces.yaml_roundtrip import YamlRoundtrip
 ModuleNotFoundError: No module named 'mpf.file_interfaces.yaml_roundtrip'

```
I searched briefly around the repo and missionpinball org on github, but couldnt find this defined anywhere. Any tips?

--- a/mpf/platforms/fast/fast.py
+++ b/mpf/platforms/fast/fast.py
@@ -676,13 +676,14 @@ class FastHardwarePlatform(ServoPlatform, LightsPlatform, RgbDmdPlatform,
                 if 0 <= channel <= 2:
                     result = []
                     for i in range(3):
+                        working_parts = parts.copy()
                         if i + channel > 2:
                             # Channel rolls over, increment the LED number
-                            parts[3] = str(int(parts[3]) + 1)
-                            parts[4] = '0'
+                            working_parts[3] = str(int(working_parts[3]) + 1)
+                            working_parts[4] = str((channel + i) % 3)
                         else:
-                            parts[4] = str(channel + i)
-                        result.append({'number': '-'.join(parts)})
+                            working_parts[4] = str(channel + i)
+                        result.append({'number': '-'.join(working_parts)})
                     return result
                 raise AssertionError(f"Invalid LED channel: {channel}")
             raise AssertionError(f"Invalid LED number: {number}")

--- a/mpf/tests/machine_files/fast/config/exp.yaml
+++ b/mpf/tests/machine_files/fast/config/exp.yaml
@@ -63,11 +63,11 @@ lights:
   led3:
     number: brian-1-3
   led4:
-    number: brian-b1-2-1
+    number: brian-b1-2-1-0
   led5:
-    number: brian-b1-2-2
+    number: brian-b1-2-2-1
   led6:
-    number: aaron-b2-1-1
+    number: aaron-b2-1-1-2
   led7:
     number: aaron-1-1
   led8:

--- a/mpf/tests/test_Fast_Exp.py
+++ b/mpf/tests/test_Fast_Exp.py
@@ -89,6 +89,12 @@ class TestFastExp(TestFastBase):
         self.assertEqual(self.led5.hw_drivers['green'][0].number, '88121-2')
         self.assertEqual(self.led5.hw_drivers['blue'][0].number, '88122-0')
 
+        # Intentionally-failing test to prove that third channel rollover has an issue
+        self.assertEqual(self.led6.hw_drivers['red'][0].number, '89200-2')
+        self.assertEqual(self.led6.hw_drivers['green'][0].number, '89201-0')
+        # Bug is generating as 202-0 instead of 201-1
+        self.assertEqual(self.led6.hw_drivers['blue'][0].number, '89201-1')
+
 
         # Make sure all the RGBW, channels, previous, and start_channels are working
         self.assertEqual(self.led22.hw_drivers['red'][0].number, '48002-0')

--- a/mpf/tests/test_Fast_Exp.py
+++ b/mpf/tests/test_Fast_Exp.py
@@ -80,6 +80,16 @@ class TestFastExp(TestFastBase):
         self.assertIn("88121", self.fast_exp_leds)
         self.assertIn("89200", self.fast_exp_leds)
 
+        # Make sure explicit offset declarations work
+        self.assertEqual(self.led4.hw_drivers['red'][0].number, '88120-0')
+        self.assertEqual(self.led4.hw_drivers['green'][0].number, '88120-1')
+        self.assertEqual(self.led4.hw_drivers['blue'][0].number, '88120-2')
+
+        self.assertEqual(self.led5.hw_drivers['red'][0].number, '88121-1')
+        self.assertEqual(self.led5.hw_drivers['green'][0].number, '88121-2')
+        self.assertEqual(self.led5.hw_drivers['blue'][0].number, '88122-0')
+
+
         # Make sure all the RGBW, channels, previous, and start_channels are working
         self.assertEqual(self.led22.hw_drivers['red'][0].number, '48002-0')
         self.assertEqual(self.led22.hw_drivers['green'][0].number, '48002-1')

--- a/mpf/tests/test_Fast_Exp.py
+++ b/mpf/tests/test_Fast_Exp.py
@@ -84,15 +84,11 @@ class TestFastExp(TestFastBase):
         self.assertEqual(self.led4.hw_drivers['red'][0].number, '88120-0')
         self.assertEqual(self.led4.hw_drivers['green'][0].number, '88120-1')
         self.assertEqual(self.led4.hw_drivers['blue'][0].number, '88120-2')
-
         self.assertEqual(self.led5.hw_drivers['red'][0].number, '88121-1')
         self.assertEqual(self.led5.hw_drivers['green'][0].number, '88121-2')
         self.assertEqual(self.led5.hw_drivers['blue'][0].number, '88122-0')
-
-        # Intentionally-failing test to prove that third channel rollover has an issue
         self.assertEqual(self.led6.hw_drivers['red'][0].number, '89200-2')
         self.assertEqual(self.led6.hw_drivers['green'][0].number, '89201-0')
-        # Bug is generating as 202-0 instead of 201-1
         self.assertEqual(self.led6.hw_drivers['blue'][0].number, '89201-1')
 
 

